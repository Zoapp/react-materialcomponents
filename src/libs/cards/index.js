/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
