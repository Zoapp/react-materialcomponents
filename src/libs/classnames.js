/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
  Rewritten to ES7 by Mik BRY
*/
const ClassNames = (args) => {
  const classes = [];
  args.forEach((arg) => {
    if (arg) {
      const argType = typeof arg;
      if (argType === "string" || argType === "number") {
        classes.push(arg);
      } else if (Array.isArray(arg) && arg.length) {
        const inner = ClassNames(arg);
        if (inner) {
          classes.push(inner);
        }
      } else if (argType === "object") {
        Object.keys(arg).forEach((key) => {
          if (arg[key]) {
            classes.push(key);
          }
        });
      }
    }
  });
  return classes.join(" ");
};

export default ClassNames;
