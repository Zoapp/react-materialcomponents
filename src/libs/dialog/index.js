/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Dialog from "./dialog";
import DialogBody from "./body";
import DialogFooter from "./footer";
import DialogHeader from "./header";

export { Dialog, DialogBody, DialogFooter, DialogHeader };
