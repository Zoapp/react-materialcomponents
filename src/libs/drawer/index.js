import Drawer from "./drawer";
import DrawerContent from "./content";
import DrawerHeader from "./header";

export { Drawer, DrawerContent, DrawerHeader };
