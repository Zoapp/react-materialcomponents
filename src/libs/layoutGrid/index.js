/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Grid from "./grid";
import Inner from "./inner";
import Cell from "./cell";

export { Grid, Inner, Cell };
