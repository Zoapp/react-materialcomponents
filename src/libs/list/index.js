/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import ListItem from "./item";
import ListItemMeta from "./meta";
import List from "./list";
import ListDivider from "./divider";
import ListGroup from "./group";
import ListGroupSubheader from "./subheader";

export {
  List,
  ListItem,
  ListItemMeta,
  ListDivider,
  ListGroup,
  ListGroupSubheader,
};
