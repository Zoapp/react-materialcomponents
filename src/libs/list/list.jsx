/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";
import Rmdc from "../";

/**
 * mdc-list-item
 * See:
 * https://material.io/components/web/catalog/lists/
 *
 */

const MDC_LIST = "mdc-list";

const List = ({
  children, dense, avatar, twoLine, ...props
}) => {
  let classes = MDC_LIST;
  if (dense) {
    classes += " mdc-list--dense";
  }
  if (avatar) {
    classes += " mdc-list--avatar-list";
  }
  if (twoLine) {
    classes += " mdc-list--two-line";
  }
  // TODO iterate througth children to check and set type
  return Rmdc.render(<ul className={classes} >{children}</ul>, props);
};

List.defaultProps = {
  mdcElement: MDC_LIST,
  children: null,
  dense: false,
  avatar: false,
  twoLine: false,
};

List.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
  dense: PropTypes.bool,
  avatar: PropTypes.bool,
  twoLine: PropTypes.bool,
};

export default List;
