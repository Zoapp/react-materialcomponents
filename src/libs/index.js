/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
