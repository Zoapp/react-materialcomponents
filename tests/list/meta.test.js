import React from "react";
import renderer from "react-test-renderer";
import { ListItemMeta } from "libs/list";

describe("list/ListItemMeta", () => {
  it("can have children", () => {
    const tree = renderer
      .create(
        <ListItemMeta>
          <div>Children are here!</div>
        </ListItemMeta>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a specified classname", () => {
    const tree = renderer.create(<ListItemMeta className="foo" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have an icon", () => {
    const tree = renderer.create(<ListItemMeta icon="favorite" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be colored", () => {
    const tree = renderer.create(<ListItemMeta color="pink" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a label", () => {
    const tree = renderer.create(<ListItemMeta label="foobar" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a link", () => {
    const tree = renderer.create(<ListItemMeta href="foobar" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
