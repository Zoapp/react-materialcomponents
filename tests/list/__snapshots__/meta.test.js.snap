// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list/ListItemMeta can be colored 1`] = `
<span
  className="mdc-list-item__meta"
  onClick={null}
  onKeyUp={[Function]}
  role="button"
  tabIndex="0"
/>
`;

exports[`list/ListItemMeta can have a label 1`] = `
<span
  className="mdc-list-item__meta"
  onClick={null}
  onKeyUp={[Function]}
  role="button"
  tabIndex="0"
/>
`;

exports[`list/ListItemMeta can have a link 1`] = `
<span
  className="mdc-list-item__meta"
  onClick={null}
  onKeyUp={[Function]}
  role="button"
  tabIndex="0"
/>
`;

exports[`list/ListItemMeta can have a specified classname 1`] = `
<span
  className="foo mdc-list-item__meta"
  onClick={null}
  onKeyUp={[Function]}
  role="button"
  tabIndex="0"
/>
`;

exports[`list/ListItemMeta can have an icon 1`] = `
<i
  className="mdc-list-item__meta material-icons"
  href={null}
>
  favorite
</i>
`;

exports[`list/ListItemMeta can have children 1`] = `
<span
  className="mdc-list-item__meta"
  onClick={null}
  onKeyUp={[Function]}
  role="button"
  tabIndex="0"
>
  <div>
    Children are here!
  </div>
</span>
`;
